zug/util.hpp

```diff
@@ -112,6 +112,24 @@ struct composed
     }
 };
 
+template <typename Fn>
+decltype(auto) to_function_tuple(Fn&& fn)
+{
+    return std::make_tuple(std::forward<Fn>(fn));
+}
+
+template <typename... Fns>
+decltype(auto) to_function_tuple(composed<Fns...> c)
```

Comment:
(I'd change this to a SFINAE approach so that the `composed` can be perfectly forwarded)